core/src/main/java/org/apache/brooklyn/util/core/flags/TypeCoercions.java

```diff
@@ -138,6 +139,9 @@ private TypeCoercions() {}
     @SuppressWarnings({ "unchecked" })
     public static <T> T coerce(Object value, TypeToken<T> targetTypeToken) {
         if (value==null) return null;
+        if (value instanceof DeferredSupplier<?>) {
+            value = ((DeferredSupplier<?>) value).get();
```

Comment:
I probably agree with this, but I worry about the implications. The `deferredSupplier.get()` call might be very slow. Are we always doing this in a context where we accept that slowness (e.g. never called by `getNonBlocking()` without first checking that it's not a deferredSupplier).

@alasdairhodge can you rebase this against master - we've fixed quite a few non-deterministic test failures, so hopefully the jenkins build will be better behaved this time.